addons/utm/models/utm_medium.py

```diff
@@ -20,10 +21,12 @@ class UtmMedium(models.Model):
 
     @api.model_create_multi
     def create(self, vals_list):
-        new_names = self.env['utm.mixin']._get_unique_names(self._name, [vals.get('name') for vals in vals_list])
-        for vals, new_name in zip(vals_list, new_names):
+        seach_domain = expression.OR([[('name', '=ilike', val['name'].lower())] for val in vals_list])
+        existing_records, new_vals_list = self.env['utm.mixin']._find_or_create_lower_record('utm.medium', vals_list, seach_domain)
+        new_names = self.env['utm.mixin']._get_unique_names(self._name, [vals.get('name') for vals in new_vals_list])
+        for vals, new_name in zip(new_vals_list, new_names):
             vals['name'] = new_name
-        return super().create(vals_list)
+        return super().create(new_vals_list) + existing_records
```

Comment:
@reth-odoo why do we have to lower the names i don't get that part ( also couldn't find a place where it lowers the lketters, but assuming it does i don't understand why we should do it, people should be able to create FaceBook right?)
also `_find_or_create_record` doesn't use ilike (perhaps you meant `_get_unique_names` ? )

Reply:
> _find_or_create_record doesn't use ilike

It was a statement for how things would be ^^
So it doesn't now, but it should. If we find any matching we say that's the one, regardless of case.
If none exists, we create it in lowercase. It ensures consistency at not extra cost.

> Should be able to create FaceBook

Then don't lowercase it but still consider it a duplicate and create a new record with `Facebook` `FaceBook [2]`
Or throw a validation error.

It's really weird to have a create return an existing record actually because the front-end, notably, will expect that create is a new record. So if you go to create your new source manually, and it already exists, all your values are changed to the ones of the original record.

Reply:
@reth-odoo i was also hesitant on create method returning already existing record, but then in UI it looks strange if i don't do it, see [this](https://www.awesomescreenshot.com/video/24972093?key=b8e3abf1146a15edf734b4ae8a789f03) video

I think you wrote a good idea and i will think about it, i have read it 100 times but my brain is exploding cuz it's overloaded, i will read it tomorrow

Reply:
@reth-odoo 

> Then don't lowercase it but still consider it a duplicate and create a new record with Facebook FaceBook [2]
Or throw a validation error.

i don't think that's right, because i think the whole point of ignoring letter cases is to have better reporting. So Facebook and Facebook [1] are basically the same thing, also Facebook and FACEBOOK

Reply:
If it's automatically created it won't happen, and if it's not people will notice that it's a duplicate thanks to the name.

But if you don't like the idea, I prefer validation error over silently returning an existing record in create